homework9/hw9task2/tests/test_context_manager.py

```diff
@@ -0,0 +1,47 @@
+import pytest
+from iterators_2.context_manager import SuppressorClass, suppressor_generator
+
+
+def test_suppressor_as_generator_index_error_positive():
+    with suppressor_generator(IndexError):
+        ["a", "b", "c"][4]
+
+
+def test_suppressor_as_generator_suppress_value_error_but_we_have_index_error_positive():
+    with pytest.raises(IndexError):
+        with suppressor_generator(ValueError):
+            ["a", "b", "c"][4]
+
+
+@pytest.mark.xfail
+def test_suppressor_as_generator_suppress_index_error_and_we_have_also_index_error_negative():
+    with pytest.raises(IndexError):
```

Comment:
Do you expect the code below raises an error?

Reply:
You mean about this lines?
`        with suppressor_generator(IndexError):`
`            [][0]`
No, wouldn`t be an error. 
Here `[][0]` will cause IndexError which is then suppressed by suppressor_generator(IndexError).
and then this test will fail  with error with `Failed: DID NOT RAISE <class 'IndexError'>`
because with pytest.raises(IndexError): IndexError didn`t raised , because it was already suppressed by suppressor_generator(IndexError).

Reply:
> No, wouldnt be an error.

This line of code explicitly says that you expect an IndexError.  Please resolve this contradiction.

Reply:
Okey, I agree that this test is not correct. When we use `with pytest.raises(IndexError):` we expect error but there is no error and test fails. We should write simple tests to check function works correct, like 
`    with suppressor_generator(IndexError):`
`        raise IndexError()`.